### ISSUE TITLE
[Framework] Different Data Backend Yield Different Validation Accuracy on ImageNet?

### ISSUE THREAD

**Issue by YuchuanTian** [NVIDIA/DeepLearningExamples#1171]
Related to **Framework(s)** 
*https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/Classification/ConvNets/main.py*

**Describe the bug**
When evaluating trained models, different backends yield different accuracies on ImageNet.

**To Reproduce**
1. Prepare trained models in main.py;
2. Set "--data-backend" as 'dali-cpu' and 'pytorch' respectively
3. Read validation accuracy printed in the terminal;
4. It seems that two data-backends yield accuracies that are quite different. For example, the official Pytorch ResNet-50 pretrained model (in the torchvision package https://download.pytorch.org/models/resnet50-19c8e357.pth ) yields accuracy 75.20% for 'dali-cpu' and 76.13% for 'pytorch'. The gap is rather large.

**Expected behavior**
Different data-backend yield nearly the same validation accuracy.

**Environment**
Please provide at least:
* Container version (e.g. pytorch:19.05-py3): Pytorch-1.8 Python 3.7.10
* GPUs in the system: (e.g. 8x Tesla V100-SXM2-16GB): Tesla P100
* CUDA driver version (e.g. 418.67): 11.4

**Comment by asulecki** [comment]
This is expected. Different dataloading pipelines have different implementations of various operations, like bilinear interpolation during resizing. This leads to accuracy gap when running validation on a databackend with which the model was not trained.

**Comment by YuchuanTian** [comment]
Though the accuracy gap seems to be a bit large, I do think your answer makes a lot of sense. Thank you very much!
